clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -128,14 +173,81 @@ public void done(long baseOffset, long logAppendTime, RuntimeException exception
         produceFuture.done();
     }
 
+    public Deque<ProducerBatch> split(int batchSize) {
```

Comment:
Also, is there a clearer name for the `batchSize` argument? e.g., `splitBatchSize` or `targetBatchSize`?